wsmaster/codenvy-hosted-api-system-license/src/main/java/com/codenvy/api/license/SystemLicenseFilter.java

```diff
@@ -0,0 +1,151 @@
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.api.license;
+
+import com.codenvy.api.license.server.SystemLicenseManager;
+import com.codenvy.api.license.server.SystemLicenseService;
+import com.codenvy.api.license.shared.dto.LegalityDto;
+import com.codenvy.api.permission.server.SystemDomain;
+import com.codenvy.auth.sso.client.filter.RequestFilter;
+import com.codenvy.auth.sso.client.filter.UriStartFromRequestFilter;
+import com.google.common.collect.ImmutableList;
+import com.google.inject.name.Named;
+import org.eclipse.che.api.core.BadRequestException;
+import org.eclipse.che.api.core.ConflictException;
+import org.eclipse.che.api.core.ForbiddenException;
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.core.UnauthorizedException;
+import org.eclipse.che.api.core.rest.HttpJsonRequestFactory;
+import org.eclipse.che.commons.env.EnvironmentContext;
+import org.eclipse.che.dto.server.DtoFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.UriBuilder;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import static com.codenvy.api.license.shared.model.Issue.Status.FAIR_SOURCE_LICENSE_IS_NOT_ACCEPTED;
+
+/**
+ * Checks system license conditions.
+ * If ("no.user.interaction" property == false) and (user has permission to perform MANAGE_SYSTEM_ACTION), then send redirection to accept-fair-source-license page in response.
+ * If ("no.user.interaction" property == true), then don't send redirection despite of whoever user is.
+ * @author Dmytro Nochevov
+ */
+@Singleton
+public class SystemLicenseFilter implements Filter {
+    public static final String NO_USER_INTERACTION                 = "no.user.interaction";
+    public static final String ACCEPT_FAIR_SOURCE_LICENSE_PAGE_URL = "license.system.accept_fair_source_license_page_url";
+
+    @Inject
+    protected RequestFilter          requestFilter;
+    @Inject
+    protected HttpJsonRequestFactory requestFactory;
+    @Inject
+    @Named("che.api")
+    protected String                 apiEndpoint;
+    @Inject
+    @Named(NO_USER_INTERACTION)
+    protected boolean                noUserInteraction;
+    @Inject
+    @Named(ACCEPT_FAIR_SOURCE_LICENSE_PAGE_URL)
```

Comment:
Looks like it should be optional because it will be unused when `noUserInteraction` is true.